lib/chainlock/chainlock.js

```diff
@@ -0,0 +1,278 @@
+    info = ChainLock.fromHex(arg);
+  } else {
+    throw new TypeError('Unrecognized argument for ChainLock');
+  }
+  return info;
+}
+
+ChainLock._fromObject = function _fromObject(data) {
+  $.checkArgument(data, 'data is required');
+  var blockHash = data.blockHash || data.blockhash;
+  var signature = data.signature;
+  if (isString(blockHash)) {
+    blockHash = BufferUtil.reverse(Buffer.from(blockHash, 'hex'));
+  }
+
+  if (isString(data.signature)) {
+    signature = Buffer.from(data.signature, 'hex');
+  }
+  var info = {
+    height: data.height,
+    blockHash: blockHash,
+    signature: signature
+  }
+  return info;
+}
+
+/**
+ * @param {BufferReader} br - Chainlock data
+ * @returns {Object} - An object representing the chainlock data
+ * @private
+ */
+ChainLock._fromBufferReader = function _fromBufferReader(br) {
+  var info = {};
+  info.height = br.readInt32LE();
+  info.blockHash = br.read(SHA256_HASH_SIZE);
+  info.signature = br.read(BLS_SIGNATURE_SIZE);
+  return info;
+};
+
+/**
+ * @param {BufferReader} br A buffer reader of the block
+ * @returns {ChainLock} - An instance of ChainLock
+ */
+ChainLock.fromBufferReader = function fromBufferReader(br) {
+  $.checkArgument(br, 'br is required');
+  var data = ChainLock._fromBufferReader(br);
+  return new ChainLock(data);
+};
+
+/**
+ * Creates ChainLock from a hex string.
+ * @param {String} string - A hex string representation of the chainLock
+ * @return {ChainLock} - An instance of ChainLock
+ */
+ChainLock.fromHex = ChainLock.fromString = function fromString(string) {
+  return ChainLock.fromBuffer(Buffer.from(string, 'hex'));
+};
+
+
+/**
+ * Creates ChainLock from a Buffer.
+ * @param {Buffer} buffer - A buffer of the chainLock
+ * @return {ChainLock} - An instance of ChainLock
+ */
+ChainLock.fromBuffer = function fromBuffer(buffer) {
+  return ChainLock.fromBufferReader(new BufferReader(buffer));
+};
+
+/**
+ * Create ChainLock from an object
+ * @param {Object} obj - an object with all properties of chainlock
+ * @return {ChainLock}
+ */
+ChainLock.fromObject = function fromObject(obj) {
+  var data = ChainLock._fromObject(obj);
+  return new ChainLock(data);
+};
+
+/**
+ * Verify that the signature is valid against the Quorum using quorumPublicKey
+ * @param {QuorumEntry} quorumEntry - quorum entry to test signature against
+ * @returns {Promise<Boolean>} - if the signature is valid for this quorum
+ */
+ChainLock.prototype.verifySignatureAgainstQuorum = async function verifySignatureAgainstQuorum(quorumEntry) {
+  var self = this;
+  return new Promise(function(resolve) {
+    var quorumPublicKey = quorumEntry.quorumPublicKey;
+    var signature = self.signature;
+    var signatureId = self.getSignatureIDForQuorumEntry(quorumEntry);
+
+    var blsInstance = bls.getInstance();
+    return blsInstance.then((instance) => {
+      var quorumPubKey = instance.PublicKey.fromBytes(Uint8Array.from(Buffer.from(quorumPublicKey, 'hex')));
```

Comment:
`Buffer` just extends `Uint8Array` so it should work too without conversion.

Reply:
Indeed, fixed!